binderhub/templates/deployment.yaml

```diff
@@ -78,6 +78,11 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        - name: JUPYTERHUB_API_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: hub-secret
```

Comment:
We should render this into a binderhub-secret rather than rely on hub-secret, since hub-secret is an implementation detail of the jupyterhub chart. There are also practical problems - this prevents binder and jupyterhub from being in different namespaces, for example.